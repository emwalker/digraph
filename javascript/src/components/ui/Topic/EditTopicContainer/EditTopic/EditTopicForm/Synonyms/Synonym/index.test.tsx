import React from 'react'
import { shallow } from 'enzyme'

import { UnwrappedSynonym as Synonym } from './index'

const onDelete = jest.fn()

const props = {
  onDelete,
  position: 0,
  synonym: {
    id: '1234',
    locale: 'en',
    name: 'Gnusto',
  },
}

describe('<Synonym />', () => {
  // @ts-expect-error
  const wrapper = shallow(<Synonym {...props} />)
  const octicon = () => wrapper.find('GoTrashcan')

  it('renders', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('when onDelete is present', () => {
    beforeEach(() => {
      wrapper.setProps({ onDelete })
    })

    it('displays the trashcan icon', () => {
      expect(octicon().exists()).toBeTruthy()
    })
  })

  describe('when onDelete is not present', () => {
    beforeEach(() => {
      wrapper.setProps({ onDelete: null })
    })

    it('displays the trashcan icon', () => {
      expect(octicon().exists()).toBeFalsy()
    })
  })
})
