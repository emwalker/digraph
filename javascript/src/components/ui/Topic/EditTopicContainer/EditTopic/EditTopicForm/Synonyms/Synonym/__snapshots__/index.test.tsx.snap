// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Synonym /> renders 1`] = `
<li
  className="Box-row clearfix css-truncate p-2 d-flex"
>
  <div
    className="col-10"
  >
    Gnusto
  </div>
  <div
    className="col-1"
  >
    en
  </div>
  <div
    className="col-1 remove-synonym"
  >
    <span
      onClick={[Function]}
      role="button"
      tabIndex={0}
    >
      <GoTrashcan />
    </span>
  </div>
</li>
`;
